tools/djxl.cc

```diff
@@ -242,8 +241,11 @@ jxl::Status WriteJxlOutput(const DecompressArgs& args, const char* file_out,
   // (Writing large PNGs is slow, so allow skipping it for benchmarks.)
   if (file_out == nullptr) return true;
 
-  // Override original color space with arg if specified.
+  // Stay in original color space unless something else is needed.
   jxl::ColorEncoding c_out = io.metadata.m.color_encoding;
+  // Override original color space with sRGB if input is CMYK.
+  if (io.Main().HasBlack()) c_out = jxl::ColorEncoding::SRGB(false);
```

Comment:
Isn't there any change required for the C API to handle this? @lvandeve

Reply:
One question to ask is: is sRGB the only option here, or can we choose others and may others be better in some cases? sRGB cannot represent all CMYK images

In the API, I don't think a change is needed to handle this (other than in the implementation, since djxl is only the decoder binary) since it already can choose and force the output color format now too

However other options to handle CMYK need to be added to the API, such as to get the 4 CMYK channels without conversion. That's different than this discussion though. The extra channels feature now can only get K.

Reply:
It's certainly not the only option - this is just what it does by default when nothing else is specified. It does have to be an RGB space if the goal is to output to any of the output formats djxl supports, and sRGB seemed like a reasonable choice, but indeed a wider gamut space that contains more of the typical CMYK gamuts might make more sense.

Getting the 4 channels without conversiom is possible in the api afaik: the color image contains CMY with 0=max ink (pretending to be RGB with 1=max light, but with an ICC profile that is CMYK), and you can get get K as an extra channel.

Reply:
It might be good to add an input/output format like TIFF that supports CMYK (currently we can use CMYK PSD input but there's no output format in djxl to get CMYK as CMYK).